### PR TITLE
Api 10199   write an oas for doc server

### PR DESCRIPTION
**Pull request by mheusgit**
Changes included in this PR

- OAS generator implemented.  Generated OAS files have been included
- Addition of new "generate:oas" script and corresponding Jest config
- Updated ReadMe to include new OAS gen instructions
- Removed no longer supported Metadata controller routes and commented out unimplemented OAS subroute
- Cleanup of Jest test.  Added new healthcheck controller test, added missing Jest test for API version controller, ~~and removed no longer supported Metadata controller test~~.
- Fixed bug with API version controller

### CODE REVIEW

File: src/api/metadata/metadata.controller.ts [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import ValidationError from '../../errors/validationError';
 import { RequestHandler, Router } from 'express';
-import apiResolverMiddleware from '../../middleware/apiResolver.middleware';
-import validationMiddleware from '../../middleware/validation.middleware';
 import APIController from '../api.controller';
-import { apiSchema } from '../api.schema';
 import APIService from '../api.service';
 import MetadataService from './metadata.service';
 
+//TODO: This class appears to be due for removal in another ticket
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/api/metadata/metadata.controller.ts]: Change was reverted

File: src/oas/oas.controller.ts [hunk 2]
```diff
@@ -99,15 +99,16 @@ class OASController {
       validationMiddleware(apiVersionLookupSchema, 'params'),
       this.createOAS,
     );
-    this.authSubRouter.put(
+    //TODO: Another ticket should implement updateOAS() then add the subrouter back
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/oas/oas.controller.ts]: Its gone

File: test/oas/oas.controller.integration.ts [hunk 3]
```diff
@@ -169,14 +171,15 @@ describe('OAS Controller', () => {
     });
 
     describe('PUT /docserver/apis/:apiName/versions/:majorVersion/oas/:oasIteration', () => {
-      it('returns with 501 unimplemented', async () => {
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply test/oas/oas.controller.integration.ts:172]: Its gone

File: test/api/metadata/metadata.controller.integration.ts [hunk 4]
```diff
@@ -1,124 +0,0 @@
...
-import { MikroORM } from 'mikro-orm';
-import supertest from 'supertest';
-import SecurityTypeEnum from '../../../src/api/security/securityType.enum';
-import APIVersionStatusEnum from '../../../src/api/version/apiVersion.status.enum';
-import configureApp from '../../../src/app';
-import { oneRingMetadata } from '../../fixtures/oneRingMetadata';
-import { initORM, wipeDatabaseRows } from '../../helpers/database';
-import { ONE_RING_OAS_URL } from '../../helpers/mockOneRingAPI';
-
-describe('MetadataController', () => {
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply test/api/metadata/metadata.controller.integration.ts]: Change was reverted except for a small addition for environment handling.

**Comment by mheusgit** [comment]
PR should be good to go

**Review by ediiotero — COMMENTED** [review]
Awesome work!
Took a quick look at the swagger UI page and looks good as far as I can tell.
Also, I agree with Christine on the commented out part of the code.

File: README.md [hunk 5]
```diff
@@ -275,6 +283,28 @@ Mermaid.js is a library to generate diagrams from a diagramming language
 3. [Class Diagrams](https://mermaid-js.github.io/mermaid/#/classDiagram)
 4. [ERD Diagrams](https://mermaid-js.github.io/mermaid/#/entityRelationshipDiagram)
 
+### DocServer's OAS
+Can be viewed through Swagger UI on DocServer at `localhost:8080/api-docs/` or on `DEV`.
```
[the remark this answers is not in the document]
Comment (christineyost) [reply README.md:287]: Sounds good

File: src/api/metadata/metadata.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (christineyost) [reply src/api/metadata/metadata.controller.ts]: This controller is used, but the corresponding table is not, and I believe the model is not

**Review by christineyost — APPROVED** [review]
Looks good to me! 🎉
I also think the Swagger UI page looks really nice!

**Review by dsteeve — COMMENTED** [review]
I think I had the same question as Christine on the OASController.

File: README.md [hunk 6]
```diff
@@ -275,6 +283,28 @@ Mermaid.js is a library to generate diagrams from a diagramming language
...
+Latest OAS can be viewed in repo at:
+  - `/docs/docserver_oas.json` OAS v2.0
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply README.md:289]: By default both get generated.  Didn't see an option to render only one.

File: src/oas/oas.controller.ts [hunk 7]
```diff
@@ -99,15 +99,6 @@ class OASController {
       validationMiddleware(apiVersionLookupSchema, 'params'),
       this.createOAS,
     );
-    this.authSubRouter.put(
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/oas/oas.controller.ts:102]: Not implemented

**Pull request merged**